docs/writing-a-reconciler.md

```diff
@@ -0,0 +1,164 @@
+# Writing a Reconciler (or Watcher)
+
+A Reconciler is a logical pattern which takes the state of a set of objects in the API server, and propagates changes to other components based on that state. 
+This logic is asnchronous, and works by having a function which is called when a change occurs in the set of resources the reconciler is watching, 
+which provides the current state of the object, and the type of change that occurred (Create, Update, Delete). 
+For more information, see the [Asynchronous Business Logic section in Platform Concepts](application-design/platform-concepts.md#asynchronous-business-logic).
+
+## Reconciler vs Watcher
```

Comment:
This section (and the considerations section) are taken mostly unchanged from the old Writing an Operator doc, with small updates where needed.